### PR TITLE
some issues in the XS code

### PR DESCRIPTION
**Pull request by mpw96**
Hi Raffael,
the issues that I'm addressing with these two commits were discovered by a coverity scan. What do you think about the fixes?

Regards
MP

### CODE REVIEW

File: SSH2.xs [hunk 1]
```diff
@@ -2048,7 +2048,7 @@ CODE:
 SSH2_CHANNEL*
 net_ls_accept(SSH2_LISTENER* ls)
 PREINIT:
-    SSH2* ss;
+    SSH2* ss = 0;
```
Comment (salva) [inline SSH2.xs]: I think this solution is wrong, you have to use the value from the `ss` slot of the structure `SSH2_LISTENER` in `ls`. Probably adding the following line in the `CODE` section:

     ss = ls->ss;

**Comment by salva** [comment]
Sorry, it seems I messed my review. The first commit should be amended. The second one is right.

File: SSH2.xs (hunk 1 above)
Comment (mpw96) [reply SSH2.xs]: well, with this change it's better than before... :-) However, I'll improve this.

**Comment by salva** [comment]
Thanks!

**Comment by salva** [comment]
I have just uploaded version 0.67 to CPAN which includes your fixes:

  https://metacpan.org/release/SALVA/Net-SSH2-0.67

**Pull request merged**